analysis/sdc.py

```diff
@@ -0,0 +1,25 @@
+"""Statistical Disclosure Control (SDC) functions.
+
+For more information, see:
+https://docs.opensafely.org/releasing-files/
+"""
+import functools
+
+
+def redact_le(series, threshold):
```

Comment:
I think it's important to separate what we could call the _intent_ of a function from what we could call the _implementation_ of a function. Here, the intent is to redact values in the given series that are less-than-or-equal-to the given threshold. The word _redact_ has meaning in our domain; we use it [in our documentation][1]. The implementation is to replace these values with zero in a copy of the given series, leaving the given series unchanged. The `redact_le` function is so named because this name expresses the intent of the function; in other words, this name has meaning in our domain.

I used `le` because [Pandas has an `le` method][2]. I decided `less_than_or_equal_to` was verbose, especially when turned into a partial function (`redact_less_than_or_equal_to_five`).

Of course, in other studies, redact may be implemented differently (it may replace these values with `pandas.NA`, for example). However, we'd know that the intention was the same: safe outputs.

[1]: https://docs.opensafely.org/releasing-files/#redacting-counts-less-than-or-equal-to-5
[2]: http://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.le.html

Reply:
thanks, that's a good point